### ISSUE TITLE
Failing to install on redhat 7

### ISSUE THREAD

**Issue by horndog** [ropensci/ijtiff#10]
Hi,

This package is failing for me when installing on Redhat 7.5.

First attempt at installing told us to install libtiff-devel. See below.

```
install.packages("/data/software/r_packages/ijtiff_1.5.1.tar.gz", lib="/usr/lib64/R/library")
inferring 'repos = NULL' from 'pkgs'
* installing *source* package ‘ijtiff’ ...
** package ‘ijtiff’ successfully unpacked and MD5 sums checked
** using staged installation
Package libtiff-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libtiff-4.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libtiff-4' found
Using PKG_CFLAGS=
Using PKG_LIBS=-ltiff -ljpeg -lz
------------------------- ANTICONF ERROR ---------------------------
Configuration failed because libtiff-4 was not found. Try installing:
 * deb: libtiff-dev (Debian, Ubuntu, etc)
 * rpm: libtiff-devel (Fedora, EPEL)
 * brew: libtiff (OSX)
If libtiff-4 is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a libtiff-4.pc file. If pkg-config
is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
--------------------------------------------------------------------
ERROR: configuration failed for package ‘ijtiff’
* removing ‘/usr/lib64/R/library/ijtiff’
Warning message:
In install.packages("/data/software/r_packages/ijtiff_1.5.1.tar.gz",  :
```

Installed that fine and now getting the following error.

```
install.packages("/data/software/r_packages/ijtiff_1.5.1.tar.gz", lib="/usr/lib64/R/library")
inferring 'repos = NULL' from 'pkgs'
* installing *source* package ‘ijtiff’ ...
** package ‘ijtiff’ successfully unpacked and MD5 sums checked
** using staged installation
Found pkg-config cflags and libs!
Using PKG_CFLAGS=
Using PKG_LIBS=-ltiff -ljbig -ljpeg -lz
** libs
g++ -m64 -std=gnu++11 -I"/usr/include/R" -DNDEBUG  -I"/usr/lib64/R/library/Rcpp/include" -I/usr/local/include  -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -c RcppExports.cpp -o RcppExports.o
gcc -m64 -std=gnu99 -I"/usr/include/R" -DNDEBUG  -I"/usr/lib64/R/library/Rcpp/include" -I/usr/local/include  -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -c common.c -o common.o
gcc -m64 -std=gnu99 -I"/usr/include/R" -DNDEBUG  -I"/usr/lib64/R/library/Rcpp/include" -I/usr/local/include  -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -c read.c -o read.o
g++ -m64 -std=gnu++11 -I"/usr/include/R" -DNDEBUG  -I"/usr/lib64/R/library/Rcpp/include" -I/usr/local/include  -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -c utils.cpp -o utils.o
gcc -m64 -std=gnu99 -I"/usr/include/R" -DNDEBUG  -I"/usr/lib64/R/library/Rcpp/include" -I/usr/local/include  -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -c write.c -o write.o
g++ -m64 -std=gnu++11 -shared -L/usr/lib64/R/lib -Wl,-z,relro -o ijtiff.so RcppExports.o common.o read.o utils.o write.o -ltiff -ljbig -ljpeg -lz -L/usr/lib64/R/lib -lR
/usr/bin/ld: cannot find -ljbig
collect2: error: ld returned 1 exit status
make: *** [ijtiff.so] Error 1
ERROR: compilation failed for package ‘ijtiff’
* removing ‘/usr/lib64/R/library/ijtiff’
Warning message:
In install.packages("/data/software/r_packages/ijtiff_1.5.1.tar.gz",  :
```
The installation page says to install libtiff5-dev but not sure that package exists on Fedora/Redhat? Can only see libtiff-devel unless that is the same.

> On Fedora Linux, try sudo yum install libtiff5-dev, or if that doesn’t work, try
> sudo yum install libtiff4-dev.

Any advice on installing would be appreciated.

Thanks

**Comment by rorynolan** [comment]
Hello,
Thanks for the heads up. Seems to me you need to install jbigkit (https://www.cl.cam.ac.uk/~mgk25/jbigkit/) (or maybe jbigkit devel). Can you try to do that and let me know how it goes? I suspect that if you manage to do that, it'll tell you you need libjpeg and libz too, but maybe you already have those. Start with jbigkit and see how you go. 
This has made me realize the installation instructions are not adequate. I'll fix them when I know what works.

**Comment by horndog** [comment]
Thanks for the quick response.

Installed jbigkit and jbigkit-devel and worked fine. 

Other two packages mentioned were already installed (libjpeg and libz-devel)

Many thanks!

**Comment by rorynolan** [comment]
Oh great news @horndog !
Can you tell me what the command to install jbigkit-devel was?
Was it `sudo yum install jbigkit-devel`?
Thanks!

**Comment by horndog** [comment]
sudo yum install jbigkit-devel

sudo yum install jbigkit

**Comment by rorynolan** [comment]
Great thanks!
One last question before I leave you in peace:
Why are you using `ijtiff` v`1.5.1` instead of the latest version? Is there something you prefer about the old version?

**Comment by horndog** [comment]
Ha good spot and question. 

I'm working in an offline environment so manually managing packages via minicran. Couldn't get the latest version to work so I download an older version from the archive in cran to see if that worked. Wondering would help me troubleshoot what was going on. That was just the last package I had tried installing so hence why I was trying with that.

Tried with the latest version there and successfully compiled in R!
